src/mondemand_backend_stats_rrd_builder.erl

```diff
@@ -252,38 +265,35 @@ maybe_create (Types, AggregatedType, File) ->
     {ok, I} -> {ok, I};
     _ ->
       case Type of
-        counter -> create_counter (File);
-        gauge -> create_gauge (File);
-        statset -> create_summary (SubType, AggregatedType, File);
-        _ -> create_counter (File) % default is counter
+        counter -> create_counter (Client, File);
+        gauge -> create_gauge (Client, File);
+        statset -> create_summary (Client, SubType, AggregatedType, File);
+        _ -> create_counter (Client, File) % default is counter
       end
   end.
 
-create_counter (File) ->
+create_rrd (Client, File, RRDType, DSMin) ->
   % creates an RRD file of 438120 bytes
-  erlrrd:create ([
-      io_lib:fwrite ("~s",[File]),
-      " --step \"60\""
-      " --start \"now - 90 days\""
-      " \"DS:value:DERIVE:900:0:U\""
-      " \"RRA:AVERAGE:0.5:1:44640\""  % 31 days of 1 minute samples
-      " \"RRA:AVERAGE:0.5:15:9600\""  % 100 days of 15 minute intervals
-      " \"RRA:AVERAGE:0.5:1440:400\"" % 400 day of 1 day intervals
-    ]).
-
-create_gauge (File) ->
+  TS = mondemand_server_util:seconds_since_epoch () - 60,
+  rrdcached_client:create ( Client, File, [
+      io_lib:fwrite ("-b ~B",[TS]),
+      " -s 60"
+      " -O",
+      io_lib:fwrite (" DS:value:~s:900:~s:U",[RRDType, DSMin]),
+      " RRA:AVERAGE:0.5:1:44640"  % 31 days of 1 minute samples
+      " RRA:AVERAGE:0.5:15:9600"  % 100 days of 15 minute intervals
+      " RRA:AVERAGE:0.5:1440:400" % 400 day of 1 day intervals
```

Comment:
The summary stats keep more data, if you look below they use 1200 days and not 400 days of 1 day intervals.  You just need to add another variable to pass in.